### PR TITLE
Change NULLFAIL => SIG_NULLFAIL.

### PR DESCRIPTION
**Pull request by richardkiss**
In src/test/script_tests.cpp the ScriptErrorDesc entries all take the form ```{SCRIPT_ERR_XXX, "XXX"}``` except for ```{ SCRIPT_ERR_SIG_NULLFAIL, "NULLFAIL" }```. This patch corrects this oversight by replacing ```NULLFAIL``` with ```SIG_NULLFAIL```.

The alternative direction is in pull request https://github.com/bitcoin/bitcoin/pull/9342

### CODE REVIEW

**Comment by richardkiss** [comment]
@paveljanik any comments on this one?

**Comment by richardkiss** [comment]
@kallewoof any thoughts?

**Comment by richardkiss** [comment]
@jl2012 It looks like your code is what I changed... do you approve of this pull request? It fixes an exception to naming, and it's helpful for me, as it makes things simpler for me importing these tests into pycoin <https://github.com/richardkiss/pycoin>, and I expect it would be helpful to others importing these tests into dynamic languages.

**Comment by jl2012** [comment]
@richardkiss I think NULLDUMMY and some other flags have the same issue. The VERIFY flags and ERR flags are just not 1-to-1

File: src/test/data/script_tests.json [hunk 1]
```diff
@@ -1501,16 +1501,16 @@
...
+    "DERSIG,SIG_NULLFAIL",
     "OK",
-    "BIP66 example 4, with DERSIG and NULLFAIL"
+    "BIP66 example 4, with DERSIG and SIG_NULLFAIL"
 ],
 [
     "0x09 0x300602010102010101",
     "0x21 0x038282263212c609d9ea2a6e3e172de238d8c39cabd5ac1ca10646e23fd5f51508 CHECKSIG NOT",
-    "DERSIG,NULLFAIL",
-    "NULLFAIL",
```
Comment (richardkiss) [inline src/test/data/script_tests.json]: There is no error code named `NULLFAIL`. It's `SIG_NULLFAIL`.

**Comment by richardkiss** [comment]
Without this change, the generated file `src/test/data/script_tests.json` has errors.

There are a couple of tests that enforce `NULLFAIL` checking (around line 1504), and subsequently fail with `SIG_NULLFAIL`. However, the JSON file suggests they should fail with the (non-existent) `NULLFAIL` error.

This reduces the usefulness of the JSON file to those who want to use it. (I realize bitcoind doesn't use it, or it would have the same problem.)

**Comment by fanquake** [comment]
Going to close this. Needs a rebase, and hasn't generated any discussion again.

**Pull request closed without merging**